ably/realtime_channel.go

```diff
@@ -455,22 +474,22 @@ func (c *RealtimeChannel) notify(msg *proto.ProtocolMessage) {
 
 func (c *RealtimeChannel) lockStartRetryAttachLoop(err error) {
 	// TODO: Move to SUSPENDED; move it to DETACHED for now.
-	c.state.set(StateChanDetached, err)
-	c.state.Unlock()
+	c.lockSetState(ChannelStateDetached, err)
+	c.mtx.Unlock()
 
 	go func() {
 		// TODO: The listener should be set before unlocking the state.
 		// Will do once we have an EventEmitter.
```

Comment:
Why do we still have this TODO when we already have event emitter?

Reply:
Fixed at e44ae34.